app/controllers/transactions_controller.rb

```diff
@@ -66,4 +105,42 @@ def add_note
   def edit_note
     @transaction.note.update(note_params) if note_params[:body].present?
   end
+
+  def transactions_filtering(start_date, end_date, transactions)
+    result = transactions.where(created_at: start_date.to_date..end_date.to_date)
+    result || []
+  end
+
+  def get_debit_transactions(categories, transactions)
+    result = []
```

Comment:
`get_debit_transactions` and `get_credit_transactions` are same. 
maybe you should create union method for this?
for example: 
  ```
def transactions_by_type(categories, transactions, debit = true)
    result = []
    categories_with_type = debit ? categories.select(&:debit) : categories.reject(&:debit)
    categories_with_type.each do |category|
      result += transactions.where(person_category_id: category.person_categories)
    end
    result
  end
```

same for `get_debit_chart_data` and `get_credit_chart_data`

Reply:
Fixed.